### PR TITLE
(#4356) NCIDS Mini Landing Page Template

### LINKED ISSUE #4356: Create Mini Landing Page Template
## As a user, I want to have NCIDS components available for my mini landing page, so that I can feature content.

The Mini Landing Page content type needs to be updated to allow for the new NCIDS page template style. The Mini Landing Page Style dropdown field will have two options: Standard/Default/something and NCIDS. (wordsmith these later)

Additionally, we are updating `usa-section` on MLPs:
1. First component on an MLP page should get:
    * Top Padding: None
    * Bottom Padding:
      * 16px (Token: 2) on desktop
      * 12px (Token: 1.5) on mobile
2. All other components in the MLP page should get:
    * Top and Bottom Padding:
      * 16px (Token: 2) on desktop
      * 12px (Token: 1.5) on mobile
3. The last component on the MLP page should get:
    * Top Padding:
      * 16px (Token: 2) on desktop
      * 12px (Token: 1.5) on mobile
    * Bottom Padding:
      * 32px (Token: 4) on desktop
      * 24px (Token: 3) on mobile
4. The NCIDS Page Title Block component will have zero (0) spacing on the bottom of it regardless of what component follows
    * This reduces the overall space between the NCIDS Page Title Block and the subsequent component by half
    * This will be a universal rule for all Mini Landing Pages using the NCIDS style 

### DoR Tasks

- [x] [Content Model](https://github.com/NCIOCPL/cgov-digital-platform/wiki/Content-Type-Mini-Landing-Page) Drafted
- [x] [Content Model](https://github.com/NCIOCPL/cgov-digital-platform/wiki/Content-Type-Mini-Landing-Page) Reviewed
  - [x] Developer Review
  - [x] IA Review
- [x] [Content Model](https://github.com/NCIOCPL/cgov-digital-platform/wiki/Content-Type-Mini-Landing-Page) Approved
  - [x]  Developer Approval
  - [ ] IA Approval
- [x] Acceptance Criteria Drafted
- [ ] Scenarios created for YAML
- [x] Acceptance Criteria Reviewed
  - [x] Developer Review
  - [x] IA Review
- [x] Acceptance Criteria Approved
  - [x] Developer Approval
  - [x] IA Approval

### DoD Tasks

- [ ] Governance Drafted
- [ ] PR Review (see PR for review and comments)
- [ ] Client Review with Demo

### Acceptance Criteria

```gherkin
Scenario: Mini Landing Page Style field displays on Mini Landing Page content type
  Given the user is logged in as an editor 
    And the user adds new content 
    And selects Mini Landing Page as the content type 
  Then the edit screen for a new Mini Landing Page content item displays
    And the Mini Landing Page Style field appears
    And the field displays under the Language field and above the Site Section field
    And the field is a dropdown
    And under the dropdown the tooltip text appears that reads "Select a template for the page"
    And the dropdown defaults to "Standard" 
    And the dropdown has the option "NCIDS"
    And the field is required 
```

### Test Content
Create the following pages to help test that the template is working as expected:

English: https://www-dev-ac.cancer.gov/test/ncids-mini-landing/template
Spanish: https://www-dev-ac.cancer.gov/espanol/test-es/ncids-mini-landing-es/template
- Add new content, select Mini Landing Page as the content type
- On the Mini Landing Page Style field, select NCIDS as the page style.
- Add a rawHTML component to the page and add "hello world" - or some generic text inside a usa-prose tag (or something else?) to demonstrate that the template is working and pulling in NCIDS styling.

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

---

### PR DESCRIPTION
**Pull request by dev-rana-publicis**
Testing Links:
* Pages with NCIDS Title Block
   * https://ncigovcdode775.prod.acquia-sites.com/research/key-initiatives/ras
   * https://ncigovcdode775.prod.acquia-sites.com/test/ncids-mini-landing/template
   * https://ncigovcdode775.prod.acquia-sites.com/espanol/test-es/ncids-mini-landing/template
* Pages without NCIDS Title Block
   * https://ncigovcdode775.prod.acquia-sites.com/test/raw-html-sample-page

Closes #4356

### CODE REVIEW

**Comment by KateMashkinaNIH** [comment]
Passed QA ( got a perfect run for once https://github.com/NCIOCPL/cgov-digital-platform-acceptance-tests/actions/runs/9715812957 )

**Comment by bryanpizzillo** [comment]
@KateMashkinaNIH could you also test analytics in the raw html blocks please?

File: docroot/profiles/custom/cgov_site/modules/custom/cgov_home_landing/config/install/core.entity_view_display.node.cgov_mini_landing.default.yml [hunk 1]
```diff
@@ -128,6 +129,13 @@ content:
     third_party_settings: {  }
     weight: 7
     region: content
+  field_mlp_page_style:
+    type: list_default
+    label: above
+    settings: {  }
+    third_party_settings: {  }
+    weight: 16
+    region: content
```
Comment (bryanpizzillo) [inline docroot/profiles/custom/cgov_site/modules/custom/cgov_home_landing/config/install/core.entity_view_display.node.cgov_mini_landing.default.yml:138]: What are the other fields here? IDK if this should set to show, or hide by default. Default would be used when we don't have a view mode defined. I don't see this showing up under that circumstance.

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/content/mini_landing/paragraph--ncids-page-title-block.html.twig [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<div class="usa-section">
```
Comment (bryanpizzillo) [inline docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/content/mini_landing/paragraph--ncids-page-title-block.html.twig:1]: Shouldn't this have a `data-eddl-landing-row`?

File: docroot/profiles/custom/cgov_site/modules/custom/cgov_home_landing/config/install/field.storage.node.field_mlp_page_style.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+field_name: field_mlp_page_style
+entity_type: node
+type: list_string
+settings:
+  allowed_values:
+    -
+      value: standard
+      label: Standard
+    -
+      value: ncids
```
Comment (bryanpizzillo) [inline docroot/profiles/custom/cgov_site/modules/custom/cgov_home_landing/config/install/field.storage.node.field_mlp_page_style.yml:17]: can you make this value `ncids_without_title`? Keep the label, but if we ever need to add others we won't be able to until the values in prod are changed. E.g. you get stuck with this value for a long time.

Comment (dev-rana-publicis) [reply docroot/profiles/custom/cgov_site/modules/custom/cgov_home_landing/config/install/field.storage.node.field_mlp_page_style.yml]: After a discussion with Bryan - we are changing the value to `ncids_default` and are keeping the label as is

**Review by sarinapadilla — APPROVED** [review]
Looks good to product!

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]

**Review by bryanpizzillo — APPROVED** [review]

**Pull request merged**